File: 1_linear.c
// Linear Search to find an element in a list

#include<stdio.h>

int linear_search(int arr[], int n, int x){
    for(int i = 0; i < n; i++){
        if(arr[i] == x)
            return i;
    }
    return 0;
}

void main(){
    int arr[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    int x;
    int n = sizeof(arr)/sizeof(arr[0]);
    printf("Enter the number you want to search for: ");
    scanf("%d", &x);

    int result = linear_search(arr, n, x);
    if(result == 0){
        printf("Element not found");
    }else{
        printf("Element found at index %d", result);
    }
}

File: 1_linear.py
# Linear Search to find an element in a list
# Linear search has a time complexity of O(N), slow for large datasets.

def linear_search(List, n):
    for i in range(len(List)):
        if(List[i] == n):
            return True
    return False

list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
x = int(input("Enter the number you want to search for: "))
if linear_search(list, x):
    print("Found")
else:
    print("Not found")